tasks/connectors/edgescan/README.md

```diff
@@ -33,6 +42,76 @@ These are some quick examples:
 | kenna_api_host     | false    | Kenna API hostname                                                           | api.us.kennasecurity.com |
 | output_directory   | false    | The task will write JSON files here (path is relative to the base directory) | output/edgescan          |
 | create_findings    | false    | The task will create findings, instead of vulnerabilities                    | false                    |
+| network_vulns      | false    | The task will include network layer vulnerabilities                          | true                     |
+| application_vulns  | false    | The task will include application layer vulnerabilities                      | true                     |
+
+## Data Mappings
+
+Edgescan assets do not map directly to Kennna assets due to Edgescan assets being more flexible in their definition.
```

Comment:
I think that there's a misunderstanding surrounding the differences between Kenna and Edgescan's respective data models.

In Kenna, an asset is a host (i.e. a single computer).

In Edgescan, an asset is a group of one or more hosts which are referred to using location specifiers (e.g. using a hostname, IP address, list of IP addresses or a block of IP addresses).

We can think of Edgescan assets as the hosts that Edgescan is configured to scan, and hosts as the systems discovered by a particular scan.

Edgescan assets and Kenna assets don't map 1:1 because they're not the same thing - the connector should be translating Edgescan _hosts_ into Kenna _assets_.

Reply:
I agree that we should be translating Edgescan hosts into Kenna assets _when possible_. And the connector is making use of host data when it is available. Host data is not present for every location/vulnerability. Host data is auto generated during just one of the many scans that edgescan offers. Therefore we must find alternative ways to group vulnerability information together for when host data is not available.

When creating Kenna assets location specifiers should be used as the edgescan equivalent in my opinion since every host **must** have a corresponding location specifier.

When creating a Kenna asset from a location specifier it will search for a corresponding host. If there is a host then, the host information will be used. If there is no host then the usual location specifier data is used. 

A vulnerability's location may also be used to create a kenna asset in the case that neither location specifier or host data can be found. If there is no host data and the location specifier was deleted, for example.

Reply:
Yes, every host _must_ have a location specifier, but not every location specifier maps to a host (e.g. a block of 65,000 IP addresses doesn't necessarily map to 65,000 active assets).

Reply:
A URL location specifier, `http://demo.testfire.net` for example, will also not have a host associated with it. Vulnerabilities found at this URL may have a location such as `http://demo.testfire.net/error_handling.jsp` or `http://demo.testfire.net/osci.jsp` associated with it. If i reverted the most recent changes we would have an asset per vulnerability, or with the changes the vulnerabilities are grouped under one asset with the original URL. See pictures below.

The fact that some empty assets are also being created seems to be irrelevant as Kenna does not display an asset unless a vulnerability is associated with it. Regardless, I've removed most of the location specifier code although, some of its data will still be used when creating an asset from a vulnerability to keep the grouping described above, and shown below.

I think this grouping is better, but let me know what you think.

Before:
![image](https://user-images.githubusercontent.com/34581180/182347868-352f47b2-4328-4323-b53b-0bb6ce95f5e8.png)
![image](https://user-images.githubusercontent.com/34581180/182347950-5e1fabb8-33f6-47ad-8879-3bf30fb11281.png)

Currently:
![image](https://user-images.githubusercontent.com/34581180/182347648-6581d8c7-2ee5-4b45-a879-52ab929f7c88.png)
![image](https://user-images.githubusercontent.com/34581180/182347715-9e3abd6d-37d8-42ea-8fae-28fe8e69aad4.png)